core/src/main/scala/com/evolutiongaming/kafka/flow/PartitionFlow.scala

```diff
@@ -252,11 +253,14 @@ object PartitionFlow {
 
     val acquire: F[PartitionFlow[F]] = init as { records =>
       for {
-        _ <- NonEmptyList.fromList(records).traverse_(processRecords)
+        cores <- Runtime[F].availableCores
```

Comment:
This will also be invoked on each incoming batch. The whole `records => ...` block is a SAM implementation of `PartitionFlow`. We need something like this: 
```scala
for {
  cores <- Runtime[F].availableCores
  _ <- init
} yield { records =>
  ...
}
```